### PR TITLE
[mod] Numerous enhancements:

### PR DESCRIPTION
**Pull request by calebj**
* Case management:
  - Centralize formatting code
  - Store raw action type
  - Centralize pretty-print (emoji) representation of actions
  - Store creation and update timestamps
  - Allow edits from admin or bot owner
  - Enable or disable case creation on a per-action basis
  - Preliminary support for action durations

* Mute:
  - add case creation (on by default for server, not channel)
  - Show channel member was muted in in modlog entry

### CODE REVIEW

**Comment by aikaterna** [comment]
With muting reporting to a mod channel, a server mute will result in an error:

```
[27/11/2016 21:15] ERROR red on_command_error 238: Exception in command 'mute server'
Traceback (most recent call last):
  File "C:\Users\user\AppData\Local\Programs\Python\Python35\lib\site-packages\discord\ext\commands\core.py", line 50, in wrapped
    ret = yield from coro(*args, **kwargs)
  File "C:\Red-DiscordBot\cogs\mod.py", line 411, in server_mute
    reason=reason)
  File "C:\Red-DiscordBot\cogs\mod.py", line 1237, in new_case
    "channel"      : channel.id,
AttributeError: 'NoneType' object has no attribute 'id'
```

**Comment by calebj** [comment]
Just rebased, please test

**Comment by calebj** [comment]
Bumping this; needs testing...

**Comment by Twentysix26** [comment]
Looks good to go. Anyone else want to give it a test?

File: cogs/mod.py [hunk 1]
```diff
@@ -204,7 +277,7 @@ def __init__(self, bot):
 
     @commands.command(no_pm=True, pass_context=True)
     @checks.admin_or_permissions(ban_members=True)
-    async def ban(self, ctx, user: discord.Member, days: int=0):
+    async def ban(self, ctx, user: discord.Member, days: int=0, *, reason: str = None):
```
Comment (Twentysix26) [inline cogs/mod.py]: I can't find a good solution for this. As it is, the user would be forced to input the days to enter a reason and I really don't like it. I could parse it manually without relying on the command ext but dunno, it seems to get messy fast between defaults, sanity checks on the days, etc. Any suggestion?

Comment (calebj) [reply cogs/mod.py]: Check to see if the first word is a number, then have the caller confirm if it's part of the reason or the number of days. If it's only a number, assume the latter.

**Pull request merged**